File: lv2/12973.py
'''
짝지어 제거하기
https://school.programmers.co.kr/learn/courses/30/lessons/12973
'''
def solution(s):
  strStack = []         # stack에 쎃아가면서 이번에 넣은 값이 이전의 값과 중복되면 pop 아니면 append 하는 방식으로 진행
  for c in s:
    if len(strStack) > 0:
      if strStack[len(strStack)-1] == c:
        strStack.pop()
        continue
    strStack.append(c)    
  return 0 if len(strStack) > 0 else 1

File: lv2_js/12973.js
`
짝지어 제거하기
https://school.programmers.co.kr/learn/courses/30/lessons/12973
`
function solution(s) {
  const stack = [];
  const sList = [...s];
  for (chr of sList) { //s의 문자를 하나씩 받아오면서
    if (stack.length > 0 && stack[stack.length - 1] === chr) { //만약 stack에 값이 있다면 마지막에 넣은 값과 지금 들어온 문자가 같은지 확인
      stack.pop() //같다면 stack의 값을 빼고
    } else {
      stack.push(chr); //다르거나 스택에 값이 없다면 stack에 push
    }
  }

  return stack.length === 0 ? 1 : 0;  //stack에 값이 남아있다면 짝지어지지 못하는 것
}